aptos-move/framework/aptos-framework/sources/native_bridge.move

```diff
@@ -0,0 +1,794 @@
+module aptos_framework::native_bridge {
+
+    use aptos_framework::account;
+    use aptos_framework::native_bridge_core;
+    use aptos_framework::native_bridge_configuration;
+    use aptos_framework::native_bridge_configuration::assert_is_caller_relayer;
+    use aptos_framework::native_bridge_store;
+    use aptos_std::smart_table;
+    use aptos_framework::ethereum;
+    use aptos_framework::ethereum::EthereumAddress;
+    use aptos_framework::event::{Self, EventHandle}; 
+    use aptos_framework::signer;
+    use aptos_framework::system_addresses;
+    #[test_only]
+    use aptos_framework::aptos_account;
+    #[test_only]
+    use aptos_framework::aptos_coin::AptosCoin;
+    #[test_only]
+    use aptos_framework::native_bridge_store::assert_valid_bridge_transfer_id;
+    #[test_only]
+    use aptos_framework::coin;
+    #[test_only]
+    use aptos_framework::ethereum::valid_eip55;
+    use std::bcs;
+    use std::vector;
+    use aptos_std::aptos_hash::keccak256;
+
+    const ETRANSFER_ALREADY_PROCESSED: u64 = 1;
+    const EINVALID_BRIDGE_TRANSFER_ID: u64 = 2;
+    const EEVENT_NOT_FOUND : u64 = 3;
+    const EINVALID_NONCE : u64 = 4;
+
+    #[event]
+    /// An event triggered upon initiating a bridge transfer
+    struct BridgeTransferInitiatedEvent has store, drop {
+        bridge_transfer_id: vector<u8>,
+        initiator: address,
+        recipient: vector<u8>,
+        amount: u64,
+        nonce: u64,
+    }
+
+    #[event]
+    /// An event triggered upon completing a bridge transfer
+    struct BridgeTransferCompletedEvent has store, drop {
+        bridge_transfer_id: vector<u8>,
+        initiator: vector<u8>,
+        recipient: address,
+        amount: u64,
+        nonce: u64,
+    }
+
+    /// This struct will store the event handles for bridge events.
+    struct BridgeEvents has key, store {
+        bridge_transfer_initiated_events: EventHandle<BridgeTransferInitiatedEvent>,
+        bridge_transfer_completed_events: EventHandle<BridgeTransferCompletedEvent>,
+    }
+
+    struct Nonce has key {
+        value: u64
+    }
+
+    /// Increment and get the current nonce  
+    fun increment_and_get_nonce(): u64 acquires Nonce {  
+        let nonce_ref = borrow_global_mut<Nonce>(@aptos_framework);  
+        nonce_ref.value = nonce_ref.value + 1;  
+        nonce_ref.value  
+    }  
+
+    /// Initializes the module and stores the `EventHandle`s in the resource.
+    public fun initialize(aptos_framework: &signer) {
+        system_addresses::assert_aptos_framework(aptos_framework);
+
+        // Ensure the nonce is not already initialized
+        assert!(
+            !exists<Nonce>(signer::address_of(aptos_framework)),
+            2
+        );
+
+        // Create the Nonce resource with an initial value of 1
+        move_to<Nonce>(aptos_framework, Nonce { 
+            value: 1 
+        });
```

Comment:
so why is the initial value of the nonce 1 and not 0?

Reply:
Thanks, yes it should be 0 initially. I changed it and will push in next commit.